lib/fathead/mdn_css/fetch.pl

```diff
@@ -32,14 +33,21 @@
 my $current_active_connections = 0;
 my $maximum_active_connections = 4;
 my @keyword_urls;
+
+#hash to store properties whose urls redirect to other properties
+my %redirect_map;
+
 queue_urls_for_download();
 
 #see http://mojolicious.org/perldoc/Mojo/IOLoop#recurring
 Mojo::IOLoop->recurring(
     0 => sub {
         for ( $current_active_connections + 1 .. $maximum_active_connections ) {
-            return ( $current_active_connections or Mojo::IOLoop->stop )
-              unless my $url = shift @keyword_urls;
+            my $url = shift @keyword_urls;
+            if(!$url) {
+                DumpFile('redirect_urls.yml', { redirects => {%redirect_map} } );
```

Comment:
No, the Dump will happen for every `tick` of the IOLoop, what you'd want to do is collect whatever you want in the loop and then write the `dump`-ing logic at the bottom, after the IOLoop code around [here](https://github.com/rasikapohankar/zeroclickinfo-fathead/blob/879b07a4040376d60ad2d8202af2656207a80250/lib/fathead/mdn_css/fetch.pl#L86). Judging from the output, the solution looks really nice :+1:.